### PR TITLE
Added gameOverScreen so user knows they have ran out of lives.

### PR DESCRIPTION
**Pull request by reid-wynkoop**
Upon seeing this screen, the amount of lives resets to 1, and amount of coins resets to 0. The Player is respawned at original starting position

### CODE REVIEW

**Review by DSF256 — CHANGES_REQUESTED** [review]
So, a couple of things...

The game seems to not reset the number of lives you have when you get a game over.  The label might say that it reset it, but once you alter your life counter in any way, it reverts back to its wrong value.  To give an example, when I got a game over once, then got two 1UP's during the next game without dying, my life counter went from (1 -> 1 -> 2), as though I had started with 0 lives, even though the label said I started with 1.  This issue does not occur if you do not get a game over.  Even if you die without getting a game over, the game still counts the number of lives correctly.

Related to this, I was able to die enough times that my life counter dipped into negative numbers, but the game would not give me a game over.  This seems to be a fencepost error; change line 164 in Player.gd from a == to a <= and you should be good.

By the way, that one glitch from earlier where you could spontaneously generate coins by jumping is gone, so that's good.

File: 2d/platformer/src/Main/Game.tscn [hunk 1]
```diff
@@ -14,3 +14,28 @@ script = ExtResource( 2 )
...
+color = Color( 0.0392157, 0.0392157, 0.0392157, 1 )
+
+[node name="GameOverLabel" type="Label" parent="InterfaceLayer/GameOverScreen"]
+margin_left = 282.0
+margin_top = 206.0
+margin_right = 519.0
+margin_bottom = 254.0
+text = "You are not good enough....
+
+I suggest getting better at the game.."
```
Comment (D-B-Cooper) [inline 2d/platformer/src/Main/Game.tscn]: These responses are not insulting enough. I suggest adding phrases that will slowly wear at the psyche of player, driving them into madness.

File: 2d/platformer/src/Actors/Player.gd [hunk 2]
```diff
@@ -158,30 +159,34 @@ func _collideWithEnemyCheck():
...
+			yield(self.deadTimer, "timeout")
+			deadMenu.close()
 			
-	deadMenu.open()
-	yield(self.deadTimer, "timeout")
-	deadMenu.close()
+			#Shows the HP bar
+			for item in get_parent().get_node("Player/UI/hpCanvasLayer/hpContainer").get_children():
+				item.visible = true
+			_resetLevel(self.lives)
```
Comment (D-B-Cooper) [inline 2d/platformer/src/Actors/Player.gd:191]: Matches what a player would expect from an in game death. Well done.

File: 2d/platformer/src/Main/Game.tscn (hunk 1 above)
Comment (reid-wynkoop) [reply 2d/platformer/src/Main/Game.tscn]: lol. Good call. I'll add a few more and randomly display one upon game over

**Comment by D-B-Cooper** [comment]
> 
> 
> So, a couple of things...
> 
> The game seems to not reset the number of lives you have when you get a game over. The label might say that it reset it, but once you alter your life counter in any way, it reverts back to its wrong value. To give an example, when I got a game over once, then got two 1UP's during the next game without dying, my life counter went from (1 -> 1 -> 2), as though I had started with 0 lives, even though the label said I started with 1. This issue does not occur if you do not get a game over. Even if you die without getting a game over, the game still counts the number of lives correctly.
> 
> Related to this, I was able to die enough times that my life counter dipped into negative numbers, but the game would not give me a game over. This seems to be a fencepost error; change line 164 in Player.gd from a == to a <= and you should be good.
> 
> By the way, that one glitch from earlier where you could spontaneously generate coins by jumping is gone, so that's good.

I was able to consistently duplicate these results in-game, please review the advice above.

**Comment by reid-wynkoop** [comment]
Lives now reset correctly The change is in commit eed0891

**Review by WilltheWys — APPROVED** [review]
Everything looks like it works good so far. good job Reid

**Comment by DSF256** [comment]
Okay, this is much better now.  I'll go ahead and merge this with the master branch.

**Pull request merged**